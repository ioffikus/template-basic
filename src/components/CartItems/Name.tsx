import React from 'react';
import { Link } from 'react-router-dom';

interface IProps {
  uid: string;
  shortDescription: string;
  name: string;
}

interface IState {}

class Copyright extends React.Component<IProps, IState> {
  render() {
    const { uid, shortDescription, name } = this.props;
    return (
      <div className="root">
        <style jsx>{`
          .root {
            font-size: 16px;
            line-height: 1.3;
            font-weight: bold;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }

          .root :global(.name) {
            color: inherit;
            white-space: nowrap;
          }
          .root :global(.code) {
            color: #b2bdcb;
            font-size: 12px;
          }
        `}</style>
        <Link to={`/product/${uid}`} className="name" title={shortDescription}>
          {name}
        </Link>
        <div className="code"># {uid}</div>
      </div>
    );
  }
}

export default Copyright;
