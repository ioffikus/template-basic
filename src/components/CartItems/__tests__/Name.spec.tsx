import React from 'react';
import { shallow } from 'enzyme';
import Name from '../Name';

describe(null, () => {
  it('render() correct', () => {
    const uid = 'dummy uid';
    const shortDescription = 'dummy shot discription';
    const name = 'dummy name';
    const wrapper = shallow(<Name uid={uid} shortDescription={shortDescription} name={name} />);
    expect(wrapper).toMatchSnapshot();
  });
});
