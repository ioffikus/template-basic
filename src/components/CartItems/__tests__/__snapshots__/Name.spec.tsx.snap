// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct 1`] = `
<div
  className="jsx-1451983618 root"
>
  <JSXStyle
    css=".root.jsx-1451983618{font-size:16px;line-height:1.3;font-weight:bold;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}.root.jsx-1451983618 .name{color:inherit;white-space:nowrap;}.root.jsx-1451983618 .code{color:#b2bdcb;font-size:12px;}"
    styleId="1451983618"
  />
  <Link
    className="name"
    replace={false}
    title="dummy shot discription"
    to="/product/dummy uid"
  >
    dummy name
  </Link>
  <div
    className="jsx-1451983618 code"
  >
    # 
    dummy uid
  </div>
</div>
`;
